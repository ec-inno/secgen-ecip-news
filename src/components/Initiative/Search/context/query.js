import React from 'react';

import { defaultLangKey } from '../../../../../languages';

const queryInit = {
  filters: {},
  status: 'ALL',
  language: defaultLangKey,
  pagination: '0/10',
};

const queryReducer = (state, action) => {
  switch (action.type) {
    case 'paginate': {
      return {
        ...state,
        pagination: action.pagination,
      };
    }

    case 'changeStatus': {
      return {
        ...state,
        pagination: queryInit.pagination,
        status: action.status,
      };
    }

    case 'changeLanguage': {
      return {
        ...state,
        pagination: queryInit.pagination,
        language: action.language,
      };
    }

    case 'changeFilter': {
      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...state.filters,
          [action.filter]: action.filterValue,
        },
      };
    }

    case 'setFilters': {
      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...state.filters,
          ...action.filters,
        },
      };
    }

    case 'removeFilter': {
      const { filters } = state;
      delete filters[action.filter];

      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...filters,
        },
      };
    }

    case 'reset':
    default: {
      return queryInit;
    }
  }
};

const Context = React.createContext();

export { queryInit, queryReducer, Context as default };
