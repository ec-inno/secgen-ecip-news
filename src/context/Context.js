import React from 'react';

const initialState = {
  hideOverlay: true,
};

export const SET_LANGUAGE_OVERLAY_VISIBILITY =
  'SET_LANGUAGE_OVERLAY_VISIBILITY';

const reducer = (state, action) => {
  switch (action.type) {
    case SET_LANGUAGE_OVERLAY_VISIBILITY: {
      return Object.assign({}, state, {
        hideOverlay: action.hideOverlay,
      });
    }

    default: {
      return initialState;
    }
  }
};

const Context = React.createContext();

const ContextProviderWithReducer = ({ children }) => {
  const [store, dispatch] = React.useReducer(reducer, initialState);

  return (
    <Context.Provider value={{ store, dispatch }}>{children}</Context.Provider>
  );
};

export { ContextProviderWithReducer, Context as default };
