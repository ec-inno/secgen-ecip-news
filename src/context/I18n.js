import React from 'react';

const I18n = React.createContext();

export default I18n;
