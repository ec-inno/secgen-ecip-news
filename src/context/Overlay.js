import React, { useContext } from 'react';

const Context = React.createContext();

const useOverlayContext = () => useContext(Context);

export { useOverlayContext, Context as default };
