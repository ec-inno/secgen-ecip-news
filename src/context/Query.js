import React, { useContext } from 'react';

import { defaultLangKey } from '../../languages';

const queryInit = {
  filters: {},
  status: 'ALL',
  language: defaultLangKey,
  pagination: '0/10',
};

const queryReducer = (state, action) => {
  switch (action.type) {
    case 'paginate': {
      return {
        ...state,
        pagination: action.pagination,
      };
    }

    case 'changeLanguage': {
      return {
        ...state,
        pagination: queryInit.pagination,
        language: action.language,
      };
    }

    case 'setFilters': {
      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...state.filters,
          ...action.filters,
        },
      };
    }

    case 'unsetFilter': {
      const { filters } = state;
      delete filters[action.filter];

      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...filters,
        },
      };
    }

    case 'reset':
    default: {
      return queryInit;
    }
  }
};

const Context = React.createContext();

const useQueryContext = () => useContext(Context);

export { queryInit, queryReducer, Context as default, useQueryContext };
