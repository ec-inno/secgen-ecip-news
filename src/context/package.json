{
  "name": "@eci/context",
  "version": "0.1.0",
  "description": "",
  "private": true,
  "dependencies": {
    "@ecl/eu-preset-website": "2.8.0",
    "react": "16.9.0"
  }
}
