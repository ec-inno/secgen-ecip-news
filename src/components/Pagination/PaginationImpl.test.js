import React from 'react';
import { render } from '@testing-library/react';

import * as QueryContext from '@eci/context/Query';
import PaginationImpl from './PaginationImpl';

describe('PaginationImpl', () => {
  it('does not render on missing input', () => {
    const { container } = render(<PaginationImpl />);
    expect(container).toMatchSnapshot();
  });

  it('does not render anything on input of zero results', () => {
    const { container } = render(<PaginationImpl itemsNumber={0} />);
    expect(container).toMatchSnapshot();
  });

  it('shows one page when less than 10', () => {
    jest.spyOn(QueryContext, 'useQueryContext').mockImplementation(() => ({
      query: {
        filters: {},
        status: 'ALL',
        language: 'en',
        pagination: '0/10',
      },
      dispachQuery: jest.fn(() => {}),
    }));

    const { container } = render(<PaginationImpl itemsNumber={9} />);
    expect(container).toMatchSnapshot();
  });

  it('shows two pages with Next button when more than 10', () => {
    jest.spyOn(QueryContext, 'useQueryContext').mockImplementation(() => ({
      query: {
        filters: {},
        status: 'ALL',
        language: 'en',
        pagination: '0/10',
      },
      dispachQuery: jest.fn(() => {}),
    }));

    const { container } = render(<PaginationImpl itemsNumber={11} />);
    expect(container).toMatchSnapshot();
  });
});
