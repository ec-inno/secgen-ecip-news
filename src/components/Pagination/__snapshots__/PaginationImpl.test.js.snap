// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationImpl does not render anything on input of zero results 1`] = `
<div>
  
</div>
`;

exports[`PaginationImpl does not render on missing input 1`] = `
<div>
  
</div>
`;

exports[`PaginationImpl shows one page when less than 10 1`] = `
<div>
  <nav
    aria-label="Browse initiatives"
    class="ecl-pagination"
  >
    <ul
      class="ecl-pagination__list"
    >
      <li
        class="ecl-pagination__item ecl-pagination__item--current"
      >
        <span
          aria-current="true"
          aria-label="Page 1"
          class="ecl-pagination__text ecl-pagination__text--summary"
        >
          1
        </span>
        <span
          aria-current="true"
          class="ecl-pagination__text ecl-pagination__text--full"
        >
          Page 1
        </span>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`PaginationImpl shows two pages with Next button when more than 10 1`] = `
<div>
  <nav
    aria-label="Browse initiatives"
    class="ecl-pagination"
  >
    <ul
      class="ecl-pagination__list"
    >
      <li
        class="ecl-pagination__item ecl-pagination__item--current"
      >
        <span
          aria-current="true"
          aria-label="Page 1"
          class="ecl-pagination__text ecl-pagination__text--summary"
        >
          1
        </span>
        <span
          aria-current="true"
          class="ecl-pagination__text ecl-pagination__text--full"
        >
          Page 1
        </span>
      </li>
      <li
        class="ecl-pagination__item"
      >
        <a
          aria-label="Go to page 2"
          class="ecl-pagination__link ecl-link ecl-link--standalone"
          href="#"
        >
          2
        </a>
      </li>
      <li
        class="ecl-pagination__item ecl-pagination__item--next"
      >
        <a
          aria-label="Go to next page"
          class="ecl-pagination__link ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after"
          href="#"
        >
          <span
            class="ecl-link__label"
          >
            Next
          </span>
          Â 
          <svg
            aria-hidden="true"
            class="ecl-link__icon ecl-icon ecl-icon--xs ecl-icon--rotate-90"
            focusable="false"
          >
            
            
            <use
              xlink:href="file#ui--corner-arrow"
            />
          </svg>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
